### PR TITLE
Fix PHP lint failure

### PR DESCRIPTION
**Pull request by ellatrix**
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by ntsekouras — APPROVED** [review]

**Comment by ellatrix** [comment]
Looks like there's `in_array` failures left. I think there was just a typo in the code.

File: lib/compat.php [hunk 1]
```diff
@@ -441,18 +441,17 @@ function gutenberg_inject_default_block_context( $args ) {
...
 
-
 			// For WordPress versions that don't support the context API.
 			if ( ! $block->context ) {
 				$block->context = array();
 			}
 
 			// Inject the post context if not done by Core.
-			$needs_post_id = empty( $block_type->uses_context ) && in_array( 'postId', $block_type->uses_context, true );
+			$needs_post_id = ! empty( $block_type->uses_context ) && in_array( 'postId', $block_type->uses_context, true );
```
Comment (ellatrix) [inline lib/compat.php:450]: @youknowriad Not sure what you intended this to be, but I'll assume it needs to be inverted to check emptiness before passing it to `in_array`.

Comment (ellatrix) [reply lib/compat.php]: This change is causing some e2e tests to fail again. The current code in master seems equivalent to always setting `$needs_post_id` to false (the array can never be both empty AND have a key), so context is never injected. I pushed a commit to check if explicitly doing that fixes tests.

File: lib/compat.php [hunk 2]
```diff
@@ -441,18 +441,17 @@ function gutenberg_inject_default_block_context( $args ) {
...
 
-
 			// For WordPress versions that don't support the context API.
 			if ( ! $block->context ) {
 				$block->context = array();
 			}
 
 			// Inject the post context if not done by Core.
-			$needs_post_id = empty( $block_type->uses_context ) && in_array( 'postId', $block_type->uses_context, true );
+			$needs_post_id = false;
```
Comment (ellatrix) [inline lib/compat.php:450]: This is explicitly stating what the current code does. Any array is never both empty and has this key. A proper fix seems to cause e2e test failures. See 5076c4d0ab5bf8ac563e7b368e88ee9778b82dd2.

**Comment by noahtallen** [comment]
closing in favor of #27011

**Pull request closed without merging**